packages/electron/src/main.ts

```diff
@@ -308,9 +310,30 @@ async function start() {
     const startUrl = process.argv.find((arg) => arg.startsWith(`${appScheme}:`));
     const path = startUrl?.replace(/\w+:(\/*)/, "");
 
-    createWindow(path);
+    const window = createWindow(path);
     createApplicationMenu();
 
+    // Lock app on suspend system event
+    powerMonitor.on("suspend", async () => {
+        console.log("suspending");
+        ipcMain.emit("electron-lock-app");
+        window.webContents.send("electron-lock-app");
+    });
+
+    // Lock app on lock system event
+    powerMonitor.on("lock-screen", async () => {
+        console.log("locking screen");
+        ipcMain.emit("electron-lock-app");
+        window.webContents.send("electron-lock-app");
+    });
+
+    // TODO: Remove this
+    setTimeout(() => {
+        console.log("fake lock");
+        // ipcMain.emit("electron-lock-app");
+        window.webContents.send("electron-lock-app");
```

Comment:
There are differences in security between these two, and I'm not sure if the `ipcMain` one is enough, but I'll have to go through that _if_ we decide to implement the `ipcRenderer` stuff in the PWA, otherwise, this process can't communicate with the app.